PACKAGE = "l3cam_ros"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

# Parameters
# gen.add(name, paramtype, level, description, default, min, max, edit_method)

# CHANGE_RGB_CAMERA_BRIGHTNESS 
gen.add("change_rgb_camera_brightness", int_t, 0, "Change RGB camera brightness", 0, -15, 15)
# CHANGE_RGB_CAMERA_CONTRAST
gen.add("change_rgb_camera_contrast", int_t, 1, "Change RGB camera constrast", 10, 0, 30)
# CHANGE_RGB_CAMERA_SATURATION
gen.add("change_rgb_camera_saturation", int_t, 2, "Change RGB camera saturation", 16, 0, 60)
# CHANGE_RGB_CAMERA_SHARPNESS
gen.add("change_rgb_camera_sharpness", int_t, 3, "Change RGB camera sharpness", 16, 0, 127)
# CHANGE_RGB_CAMERA_GAMMA
gen.add("change_rgb_camera_gamma", int_t, 4, "Change RGB camera gamma", 220, 40, 500)
# CHANGE_RGB_CAMERA_GAIN
gen.add("change_rgb_camera_gain", int_t, 5, "Change RGB camera gain", 0, 0, 63)
# ENABLE_RGB_CAMERA_AUTO_WHITE_BALANCE
gen.add("enable_rgb_camera_auto_white_balance", bool_t, 6, "Enable RGB camera auto white balance", True)
# CHANGE_RGB_CAMERA_WHITE_BALANCE
gen.add("change_rgb_camera_white_balance", int_t, 7, "Change RGB camera white balance", 5000, 1000, 10000)
# ENABLE_RGB_CAMERA_AUTO_EXPOSURE_TIME
gen.add("enable_rgb_camera_auto_exposure_time", bool_t, 8, "Enable RGB camera auto exposure time", True)
# CHANGE_RGB_CAMERA_EXPOSURE_TIME
gen.add("change_rgb_camera_exposure_time", int_t, 9, "Change RGB camera exposure time", 156, 1, 10000)

exit(gen.generate(PACKAGE, "l3cam_ros", "RgbCamera"))
